plugins/modules/identity/keycloak/keycloak_realm.py

```diff
@@ -0,0 +1,802 @@
+        reset_credentials_flow=dict(type='str', aliases=['resetCredentialsFlow']),
+        reset_password_allowed=dict(type='bool', aliases=['resetPasswordAllowed']),
+        revoke_refresh_token=dict(type='bool', aliases=['revokeRefreshToken']),
+        smtp_server=dict(type='dict', aliases=['smtpServer']),
+        ssl_required=dict(type='bool', aliases=['sslRequired']),
+        sso_session_idle_timeout=dict(type='int', aliases=['ssoSessionIdleTimeout']),
+        sso_session_idle_timeout_remember_me=dict(type='int', aliases=['ssoSessionIdleTimeoutRememberMe']),
+        sso_session_max_lifespan=dict(type='int', aliases=['ssoSessionMaxLifespan']),
+        sso_session_max_lifespan_remember_me=dict(type='int', aliases=['ssoSessionMaxLifespanRememberMe']),
+        supported_locales=dict(type='list', elements='str', aliases=['supportedLocales']),
+        user_managed_access_allowed=dict(type='bool', aliases=['userManagedAccessAllowed']),
+        verify_email=dict(type='bool', aliases=['verifyEmail']),
+        wait_increment_seconds=dict(type='int', aliases=['waitIncrementSeconds']),
+    )
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=([['id', 'realm', 'enabled']]))
+
+    result = dict(changed=False, msg='', diff={}, proposed={}, existing={}, end_state={})
+
+    # Obtain access token, initialize API
+    try:
+        connection_header = get_token(
+            base_url=module.params.get('auth_keycloak_url'),
+            validate_certs=module.params.get('validate_certs'),
+            auth_realm=module.params.get('auth_realm'),
+            client_id=module.params.get('auth_client_id'),
+            auth_username=module.params.get('auth_username'),
+            auth_password=module.params.get('auth_password'),
+            client_secret=module.params.get('auth_client_secret'),
+        )
+    except KeycloakError as e:
+        module.fail_json(msg=str(e))
+
+    kc = KeycloakAPI(module, connection_header)
+
+    realm = module.params.get('realm')
+    state = module.params.get('state')
+
+    # convert module parameters to realm representation parameters (if they belong in there)
+    realm_params = [x for x in module.params
+                    if x not in list(keycloak_argument_spec().keys()) + ['state'] and
+                    module.params.get(x) is not None]
+
+    # See whether the realm already exists in Keycloak
+    before_realm = kc.get_realm_by_id(realm=realm) or {}
+
+    # Build a proposed changeset from parameters given to this module
+    changeset = dict()
+
+    for realm_param in realm_params:
+        new_param_value = module.params.get(realm_param)
+
+        # some lists in the Keycloak API are sorted, some are not.
+        if isinstance(new_param_value, list):
+            if realm_param in ['attributes']:
```

Comment:
Why would a parameter be listed also inside the `attributes` parameter? Could you please add an example of that in the EXAMPLES block?

Reply:
To be honest, I keeped this code from keycloak_client module, ClientRepresentation (https://www.keycloak.org/docs-api/12.0/rest-api/index.html#_clientrepresentation) and RealmRepresentation (https://www.keycloak.org/docs-api/12.0/rest-api/index.html#_realmrepresentation) have an object `attributes` which is a Map without more details...

Reply:
@felixfontein yes after some tests you are right.
I think this code was originally made to have only sorted lists to facilitate debugging, because the sorted variable is put in result.
I will remove it and the code of old keycloak_client could be cleaned too.